rfcs/20200920-keras-nlp-bert.md

```diff
@@ -0,0 +1,370 @@
+# keras-nlp Transformer Encoder API
+
+| Status        | Proposed      |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Zhenyu Tan (tanzheny@google.com), Francois Chollet (fchollet@google.com), Hongkun Yu (hongkuny@google.com)|
+| **Sponsor(s)** | Mark Omernick (momernick@google.com)|
+| **Updated**   | 2020-09-21                                           |
+
+
+## Objective
+
+We aim at providing a set of Keras layers to handle Transformer-Encoder BERT-style models.
+
+## Key Benefits
+
+BERT-style Transformer-Encoders are a state-of-art technique that powers many NLP tasks:
+
+- Single sentence classification task, e.g., sentiment analysis
+- Sentence pair classification task, e.g., next sentence prediction
+- Question answering task, e.g., SQuAD
+- Single sentence tagging task, e.g., named entity recognition
+
+With this proposal, Keras users will be able to handle the tasks above with a simple API. 
+
+## Design overview
+
+This proposal builds on the assumption that inputs are lookup indices, i.e., `tf.int64` sequences.
+Tokenization is not part of this proposal but will be our immediate next step.
+
+### Classification task
+
+Case where a user want to use a pretrained BERT encoder for sentiment analysis:
+
+```python
+# Considering a imbd review dataset
```

Comment:
Not very crucial but there's typo in the comment: "Considering a imbd review dataset". It should be "Considering an imdb review dataset".